### PR TITLE
Separate siad path and data directory

### PR DESCRIPTION
**Pull request by avahowell**
See https://github.com/NebulousLabs/Nodejs-Sia/pull/26.  This pull depends on #220

### CODE REVIEW

File: js/mainjs/config.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ const Path = require('path');
 const defaultConfig = {
 	homePlugin:  'Overview',
 	siad: {
-    	path: Path.join(__dirname, '../../', 'Sia'),
+    path: Path.join(__dirname, '../..', 'Sia', 'siad'),
+    datadir: Path.join(__dirname, '../..', 'Sia'),
```
Comment (lukechampine) [inline js/mainjs/config.js]: indentation?
(should have a lint rule for this)

Comment (avahowell) [inline js/mainjs/config.js]: There's this in .jshintrc:

```
"indent"        : 4         // Specify indentation spacing
```

Kind of odd that it didn't pick that up.

File: js/mainjs/config.js [hunk 2]
```diff
@@ -6,7 +6,8 @@ const Path = require('path');
 const defaultConfig = {
 	homePlugin:  'Overview',
 	siad: {
-    	path: Path.join(__dirname, '../../', 'Sia'),
+		path: Path.join(__dirname, '../..', 'Sia', 'siad'),
+		datadir: Path.join(__dirname, '../..', 'Sia'),
```
Comment (DavidVorick) [inline js/mainjs/config.js:10]: These should use multiple params instead of `/`

File: js/rendererjs/loadingScreen.js [hunk 3]
```diff
@@ -84,7 +91,7 @@ module.exports = function(initUI) {
 			defaultPath: Path.join('../', config.path),
```
Comment (DavidVorick) [inline js/rendererjs/loadingScreen.js:91]: Same here

File: js/mainjs/config.js (hunk 2 above)
Comment (lukechampine) [inline js/mainjs/config.js:10]: also, I think `Path.resolve` is preferred over `Path.join` here, though the difference is subtle

**Pull request merged**